Mage.Sets/src/mage/cards/t/ThijarianWitness.java

```diff
@@ -0,0 +1,143 @@
+package mage.cards.t;
+
+import mage.MageInt;
+import mage.MageObjectReference;
+import mage.abilities.common.DiesCreatureTriggeredAbility;
+import mage.abilities.effects.common.ExileTargetEffect;
+import mage.abilities.effects.keyword.InvestigateEffect;
+import mage.abilities.keyword.FlashAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.SubType;
+import mage.constants.WatcherScope;
+import mage.filter.FilterPermanent;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.filter.predicate.Predicate;
+import mage.filter.predicate.mageobject.AnotherPredicate;
+import mage.game.Game;
+import mage.game.combat.CombatGroup;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.watchers.Watcher;
+
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * @author notgreat
+ */
+public final class ThijarianWitness extends CardImpl {
+
+    private static final FilterPermanent filter
+            = new FilterCreaturePermanent("another creature");
+
+    static {
+        filter.add(AnotherPredicate.instance);
+        filter.add(AttackBlockAlonePredicate.instance);
+    }
+
+    public ThijarianWitness(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{1}{G}");
+
+        this.subtype.add(SubType.ALIEN);
+        this.subtype.add(SubType.CLERIC);
+        this.power = new MageInt(0);
+        this.toughness = new MageInt(4);
+
+        // Flash
+        this.addAbility(FlashAbility.getInstance());
+
+        DiesCreatureTriggeredAbility ability = new DiesCreatureTriggeredAbility(
+                new ExileTargetEffect().setText("if it was attacking or blocking alone, exile it"),
+                false, filter, true
+        );
+        ability.addEffect(new InvestigateEffect().concatBy("and"));
+        ability.withFlavorWord("Bear Witness");
+        ability.addWatcher(new AttackingBlockingAloneWatcher());
+
+        this.addAbility(ability);
+    }
+
+    private ThijarianWitness(final ThijarianWitness card) {
+        super(card);
+    }
+
+    @Override
+    public ThijarianWitness copy() {
+        return new ThijarianWitness(this);
+    }
+}
+enum AttackBlockAlonePredicate implements Predicate<Permanent> {
+    instance;
+
+    @Override
+    public boolean apply(Permanent input, Game game) {
+        return game
+                .getState()
+                .getWatcher(AttackingBlockingAloneWatcher.class)
+                .check(new MageObjectReference(input, game), game);
+    }
+
+    @Override
+    public String toString() {
+        return "attacking or blocking alone";
+    }
+}
+class AttackingBlockingAloneWatcher extends Watcher {
```

Comment:
Must use AttackedOrBlockedThisCombatWatcher

related mtg rules
Block Alone A creature “blocks alone” if it’s the only creature declared as a blocker during the declare blockers step. A creature “is blocking alone” if it’s blocking but no other creatures are.

Reply:
As I noted above, there is a difference between "attacks/blocks alone" (tracked by AttackedOrBlockedThisCombatWatcher) and "attacking/blocking alone" which is not currently tracked.

Reply:
It must be same cause both looks to the past after (after die). Try to use standard dies trigger with condition like combat phase + single permanent in related wacher

Reply:
Well, it's different :-) Then extract watcher code to new `AttackingOrBlockingThisCombatWatcher` for the future

Full rules:

506.5. A creature attacks alone if it’s the only creature declared as an attacker during the declare attackers step. A creature is **attacking alone** if it’s attacking but no other creatures are. A creature blocks alone if it’s the only creature declared as a blocker during the declare blockers step. A **creature is blocking** alone if it’s blocking but no other creatures are.

Reply:
I recommends to look at `BlockingOrBlockedWatcher` code. It uses `REMOVED_FROM_COMBAT` event to control actual list.

Reply:
Doing some testing I'm pretty sure that watcher is broken in some situations: some involving [[Brimaz, King Of Oreskos]] and similar cards since tokens entering attacking/blocking aren't counted, others involving creatures dying without a corresponding REMOVED_FROM_COMBAT event.

However, some additional tests also found a problem in my original implementation. This new implementation is both simpler and passes an obvious test that the previous one failed at.

Reply:
Try look at this comment: https://github.com/magefree/mage/commit/43dbaf405b9bae3cc54772345db10d5c1aa79eec#r54243121

There are possible miss events REMOVED_FROM_COMBAT in some use cases

Reply:
Is there a problem with using my current method of updating the data right before any event that could kill a creature?

It feels like watching every way to potentially kill a creature and then copying the combat info (the current implementation) is easier than trying to watch every possible way combat info can change and keeping it both available at right time (as multiple creatures dying will each trigger separately and so I can't just remove them as that happens) and up-to-date.

I'm thinking to rename it to `AttackingBlockingDelayedWatcher` and document clearly that the information shouldn't be used for static abilities since it will not be updated at the right time for that. Would that be enough?

Alternatively I could just give up on Thijarian Witness, if a suitable implementation would require more engine changes than I want to do for just one card (and it seems like the way you're suggesting I do it would require that).